### PR TITLE
RavenDB-17167 release memory from blittables on evict

### PR DESCRIPTION
**Pull request by ppekrol**
### Issue link

https://issues.hibernatingrhinos.com/issue/RavenDB-17167

### Additional description

When removing DocumentInfo we should release blittables to reduce memory footprint.

### Type of change

- Bug fix

### How risky is the change?

- Moderate 

### Backward compatibility

- Not relevant

### Is it platform specific issue?

- No

### Documentation update

- No documentation update is needed 

### Testing 

- It has been verified by manual testing

### Is there any existing behavior change of other features due to this change?

- No

### UI work

- No UI work is needed

### CODE REVIEW

File: src/Raven.Client/Documents/Session/InMemoryDocumentSessionOperations.cs [hunk 1]
```diff
@@ -596,8 +596,11 @@ public void Delete(string id, string expectedChangeVector)
                         DocumentsByEntity.Remove(documentInfo.Entity);
                     }
 
-                    DocumentsById.Remove(id);
-                    changeVector = documentInfo.ChangeVector;
+                    using (documentInfo)
```
Comment (ayende) [inline src/Raven.Client/Documents/Session/InMemoryDocumentSessionOperations.cs:599]: A user may return a `BlittableDocument` and keep it around after calling `Delete()`.
Not likely to be really a scenario to be worried about, but should add that to the breaking changes list?

**Review by ayende — COMMENTED** [review]
How likely that is to be actually something meaningful?

File: src/Raven.Client/Documents/Session/InMemoryDocumentSessionOperations.cs (hunk 1 above)
Comment (ppekrol) [reply src/Raven.Client/Documents/Session/InMemoryDocumentSessionOperations.cs:599]: > How likely that is to be actually something meaningful?

Problem with long-lived sessions. Please check the YT issue.

Comment (ppekrol) [reply src/Raven.Client/Documents/Session/InMemoryDocumentSessionOperations.cs:599]: Other option is to only do that on Evict and nowhere else? Thoughts?

Comment (ayende) [reply src/Raven.Client/Documents/Session/InMemoryDocumentSessionOperations.cs:599]: Just add a breaking change note and that should be it, I think.

**Pull request merged**